From 062958ac7d3f1aa7cee0a0dbaee46612b265b1a3 Mon Sep 17 00:00:00 2001
From: gbhakteX <gurudattax.bhakte@intel.com>
Date: Mon, 23 Jul 2012 16:31:02 +0530
Subject: [PORT FROM MAIN] Audio-MW: Irregular timing shown at the last few seconds of the ALAC content local playback.

BZ: 24538

Trying to get the "mPositionTimeMediaUs" when buffer size is zero resulted in unexpected time value.
Fix: Check to get the mediatime in fillbuffer only when there is non zero buffer,
(When the input buffer is available).

Change-Id: I99f462a4b02b4088781223a4e8ee98403ec14504
Orig-Change-Id: I6291a7e96040aab9f3421ecd54fe7c89326309ba
Signed-off-by: gbhakteX <gurudattax.bhakte@intel.com>
Reviewed-on: http://android.intel.com:8080/80767
Reviewed-by: M, Arulselvan <arulselvan.m@intel.com>
Tested-by: M, Arulselvan <arulselvan.m@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 media/libstagefright/AudioPlayer.cpp |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/media/libstagefright/AudioPlayer.cpp b/media/libstagefright/AudioPlayer.cpp
index 4208019..f822c46 100644
--- a/media/libstagefright/AudioPlayer.cpp
+++ b/media/libstagefright/AudioPlayer.cpp
@@ -454,9 +454,12 @@ size_t AudioPlayer::fillBuffer(void *data, size_t size) {
                 mLatencyUs = (int64_t)mAudioTrack->latency() * 1000;
             }
 
-            CHECK(mInputBuffer->meta_data()->findInt64(
-                        kKeyTime, &mPositionTimeMediaUs));
+            if(mInputBuffer->range_length() != 0) {
+               //check for non zero buffer
+               CHECK(mInputBuffer->meta_data()->findInt64(
+                           kKeyTime, &mPositionTimeMediaUs));
 
+            }
             mPositionTimeRealUs =
                 ((mNumFramesPlayed + size_done / mFrameSize) * 1000000)
                     / mSampleRate;
-- 
1.7.4.1

